src/arm/am572x-bone-common-univ.dtsi

```diff
@@ -19,666 +19,395 @@
 
 
 	/* P8_03  (ball AB8) gpio1_24 */
-	P8_03_default_pin: pinmux_P8_03_default_pin { pinctrl-single,pins = <
-		P8_03( PIN_OUTPUT_PULLDOWN | INPUT_EN | MUX_MODE14) >; };	/* mmc3_dat6.gpio1_24 */	
-	P8_03_gpio_pin: pinmux_P8_03_gpio_pin { pinctrl-single,pins = <
-		P8_03( PIN_OUTPUT | INPUT_EN | MUX_MODE14) >; };				/* mmc3_dat6.gpio1_24 */		
-	P8_03_gpio_pu_pin: pinmux_P8_03_gpio_pu_pin { pinctrl-single,pins = <
-		P8_03( PIN_OUTPUT_PULLUP | INPUT_EN | MUX_MODE14) >; };		/* mmc3_dat6.gpio1_24 */	
-	P8_03_gpio_pd_pin: pinmux_P8_03_gpio_pd_pin { pinctrl-single,pins = <
-		P8_03( PIN_OUTPUT_PULLDOWN | INPUT_EN | MUX_MODE14) >; };	/* mmc3_dat6.gpio1_24 */	
-	P8_03_gpio_input_pin: pinmux_P8_03_gpio_input_pin { pinctrl-single,pins = <
-		P8_03( PIN_INPUT | MUX_MODE14) >; };							/* mmc3_dat6.gpio1_24 */			
+	P8_03_default_pin: pinmux_P8_03_default_pin { pinctrl-single,pins = < P8_03_DEFAULT >; };
+	P8_03_gpio_pin: pinmux_P8_03_gpio_pin { pinctrl-single,pins = < P8_03_GPIO >; };
+	P8_03_gpio_pu_pin: pinmux_P8_03_gpio_pu_pin { pinctrl-single,pins = < P8_03_GPIO_PU >; };
+	P8_03_gpio_pd_pin: pinmux_P8_03_gpio_pd_pin { pinctrl-single,pins = < P8_03_GPIO_PD >; };
```

Comment:
Is this a good idea to drop the _pin from the end of the labels and nodes? should we also drop the pinmux_ at the start of all the nodes? These things make the definition bit longer but serve no real purpose at all. 

For now, when we use `show-pins` utility we see something like this `pinmux_P9_30_gpio_pin` for the pins after making the changes we will see `P9_30_gpio` and that is good in my opinion.

Reply:
It was an odd tradition, doesn't really matter. ;)

Tradition: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/arm/boot/dts/am335x-bone-common.dtsi#n65

Tradition ignored: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/arm/boot/dts/am335x-boneblack-common.dtsi#n61

but the "/* mmc3_dat6.gpio1_24 */" at the end is actually useful...

Reply:
Okay thanks, what about the use of macros instead of doing the pin muxing here?
I was thinking of making it in such a way that it becomes easier for the end-user to switch between pin A or B of a specific header pin. if you have some suggestions on this please let me know :)

Reply:
I agree to bring back the comment at the end of the line. I don't see the point of using macros here. It seems redundant and error-prone. I'd prefer things be in one place to fix. Besides, this means they can be referenced by overlays. If it is in macros, overlays wouldn't be able to use the symbols. This saves in the .dtb size, but I don't think that is a care-about.